### PR TITLE
Initial WeatherManager refactor

### PR DESCRIPTION
**Pull request by amirroth**
This looks like a huge change, but it really isn't. The number of lines changed is inflated 3 to 4X by two syntactic changes:

- Renaming the `UtilityRoutines` namespace to `Util`.
- Renaming the `dataWeatherManager` state object to `dataWeather`. As I go through some of these things I am removing the word `Manager` (and some other words too) from module names because it adds nothing. 

The primary structural are a conversion of a whole bunch of multi-dimensional arrays of scalars to a much smaller number of multi-dimensional arrays of structures. A potential follow-on clean-up pass is to use these structures more intelligently, e.g., as parameters to routines, and to reduce structure copies.

### CODE REVIEW

File: src/EnergyPlus/AirLoopHVACDOAS.cc
Comment (amirroth) [inline src/EnergyPlus/AirLoopHVACDOAS.cc:201]: First syntactic change: this is probably about 50% of all line changes. Maybe more.

File: src/EnergyPlus/AirflowNetwork/include/AirflowNetwork/Properties.hpp
Comment (amirroth) [inline src/EnergyPlus/AirflowNetwork/include/AirflowNetwork/Properties.hpp:61]: Took the opportunity to fix this annoying nonsense.

File: src/EnergyPlus/Coils/CoilCoolingDX.cc
Comment (amirroth) [inline src/EnergyPlus/Coils/CoilCoolingDX.cc:88]: Don't capitalize the same string over and over inside a loop, peeps.

File: src/EnergyPlus/DataRuntimeLanguage.hh
Comment (amirroth) [inline src/EnergyPlus/DataRuntimeLanguage.hh:457]: Snuck this set of changes in here for some reason too. You're welcome.

File: src/EnergyPlus/WeatherManager.cc
Comment (amirroth) [inline src/EnergyPlus/WeatherManager.cc:93]: The name `Manager` doesn't add anything. This isn't an org-chart.

File: src/EnergyPlus/WeatherManager.cc
Comment (amirroth) [inline src/EnergyPlus/WeatherManager.cc]: I'm making the executive decision to use untype-checked casts to upcast `enum` to `int`. Fight me.

File: src/EnergyPlus/WeatherManager.cc
Comment (amirroth) [inline src/EnergyPlus/WeatherManager.cc:192]: Use these wrappers for common errors. See `UtilityRoutines` for available wrappers.

File: src/EnergyPlus/WeatherManager.cc
Comment (amirroth) [inline src/EnergyPlus/WeatherManager.cc]: Get used to seeing this idiom, i.e., assigning within an `if` condition.

File: src/EnergyPlus/WeatherManager.cc
Comment (amirroth) [inline src/EnergyPlus/WeatherManager.cc:1933]: Here are some actual structural changes. These have been collected into structs. A future refactor could eliminate this and other copes using pointers/references.

File: src/EnergyPlus/WeatherManager.cc
Comment (amirroth) [inline src/EnergyPlus/WeatherManager.cc:2096]: These have been converted from individual arrays to arrays of structs where each struct represents all of the conditions for the time step.

File: src/EnergyPlus/WeatherManager.cc
Comment (amirroth) [inline src/EnergyPlus/WeatherManager.cc:2384]: This idiom where we pass a bunch of strings to a function (not to mention call the function to begin with) for the rare case of errors has to stop.

File: src/EnergyPlus/WeatherManager.cc
Comment (amirroth) [inline src/EnergyPlus/WeatherManager.cc:4040]: `{x,y,z}` coordinates should be `Vector3` (which is a struct) not `Array1D` (which is a heap vector).

File: src/EnergyPlus/WeatherManager.cc
Comment (amirroth) [inline src/EnergyPlus/WeatherManager.cc:4636]: I guess I should delete these.

File: src/EnergyPlus/WeatherManager.hh
Comment (amirroth) [inline src/EnergyPlus/WeatherManager.hh]: New constructor idiom.

File: src/EnergyPlus/WeatherManager.hh
Comment (amirroth) [inline src/EnergyPlus/WeatherManager.hh]: Each of these replace 16 scalar arrays.

File: src/EnergyPlus/WeatherManager.hh
Comment (amirroth) [inline src/EnergyPlus/WeatherManager.hh]: Thse should probably be made into `EPVector` as well.

File: src/EnergyPlus/WeatherManager.hh
Comment (amirroth) [inline src/EnergyPlus/WeatherManager.hh]: Will deal with this nonsense later.

**Review by amirroth — COMMENTED** [review]
Quick walkthrough. Pretty simple stuff, just many copies of it.

File: src/EnergyPlus/WeatherManager.cc
Comment (dareumnam) [reply src/EnergyPlus/WeatherManager.cc]: I think this is a very minor thing, but I see two spaces between `else` and `if`.

**Review by dareumnam — COMMENTED** [review]
This looks great to me. I've only had one minor comment. Since there is no functional change, I think it's good to go as long as the CI results are all green and the clang-format is cleaned up.

File: src/EnergyPlus/BranchNodeConnections.cc
Comment (rraustad) [inline src/EnergyPlus/BranchNodeConnections.cc:1969]: Just curious how these line feeds got added through global search and replace? I've seen several inconsequential but this one adds the line feed after the `=` instead of others that were added after the replacement line (e.g., line 1972 below).

File: src/EnergyPlus/DaylightingManager.cc
Comment (rraustad) [inline src/EnergyPlus/DaylightingManager.cc:4610]: Why not also delete the ipsc reference at line 4581 above?

File: src/EnergyPlus/BranchNodeConnections.cc
Comment (amirroth) [reply src/EnergyPlus/BranchNodeConnections.cc:1969]: Probably because I did this one by hand because it came from a merge conflict.

File: src/EnergyPlus/DaylightingManager.cc
Comment (amirroth) [reply src/EnergyPlus/DaylightingManager.cc:4610]: This branch has been ready since August but I didn't merge it because I didn't want to randomize the code-base while other people were getting their actual features and bug fixes in. As a result there were a lot more merge conflicts than usual, and this is probably the result of one of them.

File: src/EnergyPlus/AirflowNetwork/src/Solver.cpp
Comment (JasonGlazer) [inline src/EnergyPlus/AirflowNetwork/src/Solver.cpp:1377]: If I remember correctly SameString already has makeUpper in it so this makeUpper is unnecessary

File: src/EnergyPlus/FluidProperties.cc
Comment (JasonGlazer) [inline src/EnergyPlus/FluidProperties.cc:7737]: Is it necessary to do the makeUpper before FindItemInList?

File: src/EnergyPlus/FuelCellElectricGenerator.cc
Comment (JasonGlazer) [inline src/EnergyPlus/FuelCellElectricGenerator.cc:130]: Use SameString

File: src/EnergyPlus/General.cc
Comment (JasonGlazer) [inline src/EnergyPlus/General.cc:460]: Aren't there better functions to convert a string to float?

File: src/EnergyPlus/InputProcessing/InputProcessor.cc
Comment (JasonGlazer) [inline src/EnergyPlus/InputProcessing/InputProcessor.cc:1282]: Was this unused?

File: src/EnergyPlus/RuntimeLanguageProcessor.cc
Comment (JasonGlazer) [inline src/EnergyPlus/RuntimeLanguageProcessor.cc:2521]: nice changes

File: src/EnergyPlus/TarcogShading.cc
Comment (JasonGlazer) [inline src/EnergyPlus/TarcogShading.cc:709]: Huh? maybe just T0=Constant:Kelvin

File: src/EnergyPlus/TarcogShading.cc
Comment (JasonGlazer) [inline src/EnergyPlus/TarcogShading.cc:1031]: Same here

File: src/EnergyPlus/UtilityRoutines.hh
Comment (JasonGlazer) [inline src/EnergyPlus/UtilityRoutines.hh]: I'm not sure I'm following this change. are they being used?

**Review by JasonGlazer — COMMENTED** [review]
It was difficult to review since it included many non-substantive changes and then some substantive changes too. 

In general, EnergyPlus does too much string handling for a "scientific" program and the change to Util really shows that. Perhaps in the future we can do a refactor that just builds up a symbol table like a compiler and all the various compares can be between integers.

File: src/EnergyPlus/TarcogShading.cc
Comment (jasondegraw) [reply src/EnergyPlus/TarcogShading.cc:709]: I'm pretty sure that the use of the conversion factor predates a lot of what is possible now with `constexpr` and friends. Maybe we could just define a function to do the temperature conversion at compile time for stuff like this now? And ditch the constant.

File: src/EnergyPlus/InputProcessing/InputProcessor.cc
Comment (amirroth) [reply src/EnergyPlus/InputProcessing/InputProcessor.cc:1282]: It's unused now. It's a terrible idiom.

File: src/EnergyPlus/TarcogShading.cc
Comment (amirroth) [reply src/EnergyPlus/TarcogShading.cc:709]: This is all global search-replace stuff. I don't have opinions on these individual lines.

File: src/EnergyPlus/UtilityRoutines.hh
Comment (amirroth) [reply src/EnergyPlus/UtilityRoutines.hh]: Yes, I've been using them in my previous four or five PRs. Do a search on a search on some of these.

File: src/EnergyPlus/General.cc
Comment (Myoldmopar) [reply src/EnergyPlus/General.cc:460]: I agree, but that's probably not necessary here in this pass.

File: src/EnergyPlus/FluidProperties.cc
Comment (Myoldmopar) [reply src/EnergyPlus/FluidProperties.cc:7737]: FindItemInList does not explicitly call an upper case function on the inputs, so from a perspective of that function alone, you should call MakeUPPER.  However, given the layers of EnergyPlus, it's likely that things are already upper case at this point.  I'm not sure we should count on that moving forward, and leave this in place for now.

File: src/EnergyPlus/FuelCellElectricGenerator.cc
Comment (Myoldmopar) [reply src/EnergyPlus/FuelCellElectricGenerator.cc:130]: Yeah, or even just a direct call to equali which is all that SameString does anyway.  This was just a byproduct of a broad search/replace on the namespace change, so I wouldn't demand holding this up for it right now.

Comment (JasonGlazer) [reply src/EnergyPlus/FuelCellElectricGenerator.cc:130]: I understand it is a little outside the scope of this change but it made me cringe a little :-)

File: src/EnergyPlus/FluidProperties.cc
Comment (JasonGlazer) [reply src/EnergyPlus/FluidProperties.cc:7737]: I think it is FindItem not FindItemInList that does the MakeUPPER inside.

File: src/EnergyPlus/FuelCellElectricGenerator.cc
Comment (Myoldmopar) [reply src/EnergyPlus/FuelCellElectricGenerator.cc:130]: Fair, I already have to apply Clang Format, I might clean this up just to reduce your cringe! :rofl:

Comment (amirroth) [reply src/EnergyPlus/FuelCellElectricGenerator.cc:130]: You can just open up an issue to this effect also. Then in a future clean-up pass, you can take `MakeUPPER` out of all of the `FindItem` variants, see what breaks and fix it. Maybe some `assert` statements will help as well.

Comment (Myoldmopar) [reply src/EnergyPlus/FuelCellElectricGenerator.cc:130]: Yeah, good idea.  I'll hold off from anything here, just run Clang Format and get this merged if everything is clean.

**Comment by Myoldmopar** [comment]
CI is all happy, just the spurious CI diff.  I'll apply Clang Format and get this merged.  We can follow-up with any number of PRs to cleanup the same string stuff.

**Comment by Myoldmopar** [comment]
CI is completely clean running locally after pulling develop and applying Clang Format.  I'm going to push that and quickly merge.  Thanks @amirroth !

**Review by Myoldmopar — APPROVED** [review]
Hey this is great, And even 2200 lines of code removed!

**Pull request merged**
